src/test/java/seedu/address/logic/commands/DeleteGoodPricePairFromSupplierCommandTest.java

```diff
@@ -0,0 +1,222 @@
+                .concat(ALICE.getOffers().iterator().next().getGoodName().toString())
+                .concat(", ");
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), getTypicalInventory(),
+                getTypicalTransactionHistory(), new UserPrefs());
+
+        Supplier editedSupplier = new SupplierBuilder(supplierInFilteredList).withName(ALICE.getName().toString())
+                .withAddress(ALICE.getAddress().toString())
+                .withEmail(ALICE.getEmail().toString())
+                .withPhone(ALICE.getPhone().toString()).withOffers().build();
+
+        expectedModel.setSupplier(model.getFilteredSupplierList().get(0), editedSupplier);
+
+        assertCommandSuccess(deleteGoodPricePairFromSupplierCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_couldNotFindGood_success() {
+        Set<GoodName> goodNames = new HashSet<>();
+        goodNames.add(new GoodBuilder().withGoodName("There is no such good").build().getGoodName());
+
+        DeleteGoodPricePairFromSupplierCommand.DeleteSupplierGoodName descriptor =
+                new DeleteSupplierGoodNameBuilder().withGoodNames(goodNames).build();
+
+        DeleteGoodPricePairFromSupplierCommand deleteGoodPricePairFromSupplierCommand =
+                new DeleteGoodPricePairFromSupplierCommand(INDEX_FIRST_SUPPLIER, descriptor);
+
+        String expectedMessage = DeleteGoodPricePairFromSupplierCommand.MESSAGE_COULD_NOT_FIND_GOOD
+                .concat("There is no such good").concat(", ");
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), getTypicalInventory(),
+                getTypicalTransactionHistory(), new UserPrefs());
+
+        assertCommandSuccess(deleteGoodPricePairFromSupplierCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_mixSuccessfulDeleteGoodAndCouldNotFindGood_success() {
+        Set<GoodName> goodNames = new HashSet<>();
+        ALICE.getOffers().forEach(offer -> goodNames.add(offer.getGoodName()));
+        goodNames.add(new GoodBuilder().withGoodName("There is no such good").build().getGoodName());
+
+        DeleteGoodPricePairFromSupplierCommand.DeleteSupplierGoodName descriptor =
+                new DeleteSupplierGoodNameBuilder().withGoodNames(goodNames).build();
+
+        DeleteGoodPricePairFromSupplierCommand deleteGoodPricePairFromSupplierCommand =
+                new DeleteGoodPricePairFromSupplierCommand(INDEX_FIRST_SUPPLIER, descriptor);
+
+        String expectedMessage = DeleteGoodPricePairFromSupplierCommand.MESSAGE_SUCCESS_DELETE_GOOD
+                .concat(ALICE.getOffers().iterator().next().getGoodName().toString())
+                .concat(", \n").concat(DeleteGoodPricePairFromSupplierCommand.MESSAGE_COULD_NOT_FIND_GOOD)
+                .concat("There is no such good, ");
```

Comment:
Can you change the good name to something that would not be mistaken as part of the error message? Maybe make `final String NONEXISTENT_GOOD_NAME = "There is no such good"`?